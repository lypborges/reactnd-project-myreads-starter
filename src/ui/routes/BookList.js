import React, { Component } from "react";
import BookShelf from "../components/BookShelf";
import Loading from "../components/Loading";
import { Link } from "react-router-dom";
import { Layout, Icon } from "antd";
import "./BookList.css";

const { Header, Content } = Layout;

class BookList extends Component {
  _groupShelf(books, term) {
    return books.filter(book => {
      return book.shelf === term;
    });
  }

  _buildShelfs = books => {
    const currentlyReading = this._groupShelf(books, "currentlyReading");
    const wantToRead = this._groupShelf(books, "wantToRead");
    const read = this._groupShelf(books, "read");

    this.setState({
      books,
      currentlyReading,
      wantToRead,
      read,
      isLoading: false
    });
  };

  render(props) {
    const { books, isLoading } = this.props;

    const currentlyReading = this._groupShelf(books, "currentlyReading");
    const wantToRead = this._groupShelf(books, "wantToRead");
    const read = this._groupShelf(books, "read");

    return (
      <Layout>
        <Header className="main-header">
          <Icon type="book" style={{ fontSize: 50, color: "white" }} />
          <h1 className="main-title">My reads</h1>
        </Header>
        <Content className="main-content">
          {isLoading && <Loading />}
          {!isLoading && (
            <div className="list-books">
              <div className="list-books-content">
                <div>
                  <BookShelf
                    books={currentlyReading}
                    title="Currently Reading"
                    handleShelfChange={this.props.handleShelfChange}
                  />
                  <BookShelf
                    books={wantToRead}
                    title="Want to Read"
                    handleShelfChange={this.props.handleShelfChange}
                  />
                  <BookShelf
                    books={read}
                    title="Read"
                    handleShelfChange={this.props.handleShelfChange}
                  />
                </div>
              </div>
              <div className="open-search">
                <Link to="/search">
                  <Icon type="plus-circle" className="add-book-icon" />
                </Link>
              </div>
            </div>
          )}
        </Content>
      </Layout>
    );
  }
}

export default BookList;
