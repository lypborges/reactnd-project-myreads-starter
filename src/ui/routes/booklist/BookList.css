.main-content {
  margin-top: 80px;
}

.main-header {
  position: fixed;
  display: flex;
  justify-content: flex-start;
  z-index: 2;
  width: 100%;
}

.main-title {
  color: #fff;
  text-align: left;
}

.bookshelf-books {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.bookshelf-book {
  margin: 10px;
}

.bookshelf-container {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 20px;
}

.bookshelf-title {
  text-align: center;
}

.filter-book-shelf {
  display: flex;
  justify-content: center;
}

.list-spinner {
  display: flex;
  justify-content: center;
  align-content: center;
  height: 100vh;
  text-align: center;
  border-radius: 4px;
}

.add-book-icon {
  position: fixed;
  bottom: 10px;
  right: 10px;
  font-size: 50px;
}
