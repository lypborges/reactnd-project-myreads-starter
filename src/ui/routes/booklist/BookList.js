import React, { Component } from "react";
import BookShelf from "../../components/bookshelf/BookShelf";
import Loading from "../../components/loading/Loading";
import { Link } from "react-router-dom";
import { Layout, Icon } from "antd";
import "./BookList.css";

const { Header, Content } = Layout;

const ShelfContainer = props => {
  return (
    <div className="list-books">
      <div className="list-books-content">
        {props.children}
        <div className="open-search">
          <Link to="/search">
            <Icon type="plus-circle" className="add-book-icon" />
          </Link>
        </div>
      </div>
    </div>
  );
};

const Shelfs = props => {
  return (
    <ShelfContainer>
      <div>
        <BookShelf
          books={props.currentlyReading}
          title="Currently Reading"
          handleShelfChange={props.handleShelfChange}
        />
        <BookShelf
          books={props.wantToRead}
          title="Want to Read"
          handleShelfChange={props.handleShelfChange}
        />
        <BookShelf
          books={props.read}
          title="Read"
          handleShelfChange={props.handleShelfChange}
        />
      </div>
    </ShelfContainer>
  );
};

class BookList extends Component {
  _groupShelf(books, term) {
    return books.filter(book => {
      return book.shelf === term;
    });
  }

  _buildShelfs = books => {
    const currentlyReading = this._groupShelf(books, "currentlyReading");
    const wantToRead = this._groupShelf(books, "wantToRead");
    const read = this._groupShelf(books, "read");

    this.setState({
      books,
      currentlyReading,
      wantToRead,
      read,
      isLoading: false
    });
  };

  render(props) {
    const { books, isLoading, handleShelfChange } = this.props;
    const currentlyReading = this._groupShelf(books, "currentlyReading");
    const wantToRead = this._groupShelf(books, "wantToRead");
    const read = this._groupShelf(books, "read");

    return (
      <Layout>
        <Header className="main-header">
          <Icon type="book" style={{ fontSize: 50, color: "white" }} />
          <h1 className="main-title">My reads</h1>
        </Header>
        <Content className="main-content">
          {isLoading && <Loading />}
          {!isLoading && (
            <Shelfs
              currentlyReading={currentlyReading}
              wantToRead={wantToRead}
              read={read}
              handleShelfChange={handleShelfChange}
            />
          )}
        </Content>
      </Layout>
    );
  }
}

export default BookList;
