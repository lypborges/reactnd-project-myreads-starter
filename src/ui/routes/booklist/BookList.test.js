import BookList from "./BookList";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { BooksData } from "../../../tests/helpers/BookStore";

describe("Loading component layout", () => {
  test("render with books", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <BookList books={BooksData} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render without books", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <BookList />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
