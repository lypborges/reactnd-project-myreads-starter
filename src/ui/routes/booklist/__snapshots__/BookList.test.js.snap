// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component layout render with books 1`] = `
<div
  className="ant-layout"
>
  <div
    className="main-header ant-layout-header"
  >
    <i
      className="anticon anticon-book"
      style={
        Object {
          "color": "white",
          "fontSize": 50,
        }
      }
    />
    <h1
      className="main-title"
    >
      My reads
    </h1>
  </div>
  <div
    className="main-content ant-layout-content"
  >
    <div
      className="list-books"
    >
      <div
        className="list-books-content"
      >
        <div>
          <div
            className="bookshelf"
          >
            <h2
              className="bookshelf-title"
            >
              Currently Reading
            </h2>
            <div
              className="bookshelf-books"
              data-testid="shelf Currently Reading"
            >
              <div
                className="bookshelf-container"
              >
                <div
                  className="ant-card bookshelf-book ant-card-bordered"
                  style={
                    Object {
                      "width": 250,
                    }
                  }
                >
                  <div
                    className="ant-card-cover"
                  >
                    <img
                      alt="The Linux Command Line"
                      src="http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                      style={
                        Object {
                          "height": 193,
                          "width": 248,
                        }
                      }
                    />
                  </div>
                  <div
                    className="ant-card-body"
                    style={Object {}}
                  >
                    <div
                      className="ant-card-meta"
                    >
                      <div
                        className="ant-card-meta-detail"
                      >
                        <div
                          className="ant-card-meta-title"
                        >
                          The Linux Command Line
                        </div>
                        <div
                          className="ant-card-meta-description"
                        >
                          <span>
                            William E. Shotts, Jr.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    className="ant-radio-group ant-radio-group-small"
                    id={undefined}
                    onMouseEnter={undefined}
                    onMouseLeave={undefined}
                    style={undefined}
                  >
                    <label
                      className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
                      onMouseEnter={undefined}
                      onMouseLeave={undefined}
                      style={undefined}
                    >
                      <span
                        className="ant-radio-button ant-radio-button-checked"
                        style={Object {}}
                      >
                        <input
                          autoFocus={undefined}
                          checked={true}
                          className="ant-radio-button-input"
                          data-testid="button-currently-reading"
                          disabled={false}
                          id={undefined}
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={undefined}
                          onFocus={[Function]}
                          readOnly={undefined}
                          tabIndex={undefined}
                          type="radio"
                          value="currentlyReading"
                        />
                        <span
                          className="ant-radio-button-inner"
                        />
                      </span>
                      <span>
                        Reading
                      </span>
                    </label>
                    <label
                      className="ant-radio-button-wrapper"
                      onMouseEnter={undefined}
                      onMouseLeave={undefined}
                      style={undefined}
                    >
                      <span
                        className="ant-radio-button"
                        style={Object {}}
                      >
                        <input
                          autoFocus={undefined}
                          checked={false}
                          className="ant-radio-button-input"
                          data-testid="button-want-to-read"
                          disabled={false}
                          id={undefined}
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={undefined}
                          onFocus={[Function]}
                          readOnly={undefined}
                          tabIndex={undefined}
                          type="radio"
                          value="wantToRead"
                        />
                        <span
                          className="ant-radio-button-inner"
                        />
                      </span>
                      <span>
                        Want to Read
                      </span>
                    </label>
                    <label
                      className="ant-radio-button-wrapper"
                      onMouseEnter={undefined}
                      onMouseLeave={undefined}
                      style={undefined}
                    >
                      <span
                        className="ant-radio-button"
                        style={Object {}}
                      >
                        <input
                          autoFocus={undefined}
                          checked={false}
                          className="ant-radio-button-input"
                          data-testid="button-read"
                          disabled={false}
                          id={undefined}
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={undefined}
                          onFocus={[Function]}
                          readOnly={undefined}
                          tabIndex={undefined}
                          type="radio"
                          value="read"
                        />
                        <span
                          className="ant-radio-button-inner"
                        />
                      </span>
                      <span>
                        Read
                      </span>
                    </label>
                    <label
                      className="ant-radio-button-wrapper"
                      onMouseEnter={undefined}
                      onMouseLeave={undefined}
                      style={undefined}
                    >
                      <span
                        className="ant-radio-button"
                        style={Object {}}
                      >
                        <input
                          autoFocus={undefined}
                          checked={false}
                          className="ant-radio-button-input"
                          data-testid="button-none"
                          disabled={false}
                          id={undefined}
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={undefined}
                          onFocus={[Function]}
                          readOnly={undefined}
                          tabIndex={undefined}
                          type="radio"
                          value="none"
                        />
                        <span
                          className="ant-radio-button-inner"
                        />
                      </span>
                      <span>
                        None
                      </span>
                    </label>
                  </div>
                </div>
              </div>
              <div
                className="ant-divider ant-divider-horizontal"
              />
            </div>
          </div>
          <div
            className="bookshelf"
          >
            <h2
              className="bookshelf-title"
            >
              Want to Read
            </h2>
            <div>
              <h3
                className="empty-shelf"
              >
                Empty shelf ;(
              </h3>
              <div
                className="ant-divider ant-divider-horizontal"
              />
            </div>
          </div>
          <div
            className="bookshelf"
          >
            <h2
              className="bookshelf-title"
            >
              Read
            </h2>
            <div>
              <h3
                className="empty-shelf"
              >
                Empty shelf ;(
              </h3>
              <div
                className="ant-divider ant-divider-horizontal"
              />
            </div>
          </div>
        </div>
        <div
          className="open-search"
        >
          <a
            href="/search"
            onClick={[Function]}
          >
            <i
              className="anticon anticon-plus-circle add-book-icon"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Loading component layout render without books 1`] = `
<div
  className="ant-layout"
>
  <div
    className="main-header ant-layout-header"
  >
    <i
      className="anticon anticon-book"
      style={
        Object {
          "color": "white",
          "fontSize": 50,
        }
      }
    />
    <h1
      className="main-title"
    >
      My reads
    </h1>
  </div>
  <div
    className="main-content ant-layout-content"
  >
    <div
      className="list-books"
    >
      <div
        className="list-books-content"
      >
        <div>
          <div
            className="bookshelf"
          >
            <h2
              className="bookshelf-title"
            >
              Currently Reading
            </h2>
            <div>
              <h3
                className="empty-shelf"
              >
                Empty shelf ;(
              </h3>
              <div
                className="ant-divider ant-divider-horizontal"
              />
            </div>
          </div>
          <div
            className="bookshelf"
          >
            <h2
              className="bookshelf-title"
            >
              Want to Read
            </h2>
            <div>
              <h3
                className="empty-shelf"
              >
                Empty shelf ;(
              </h3>
              <div
                className="ant-divider ant-divider-horizontal"
              />
            </div>
          </div>
          <div
            className="bookshelf"
          >
            <h2
              className="bookshelf-title"
            >
              Read
            </h2>
            <div>
              <h3
                className="empty-shelf"
              >
                Empty shelf ;(
              </h3>
              <div
                className="ant-divider ant-divider-horizontal"
              />
            </div>
          </div>
        </div>
        <div
          className="open-search"
        >
          <a
            href="/search"
            onClick={[Function]}
          >
            <i
              className="anticon anticon-plus-circle add-book-icon"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
